### ISSUE TITLE
test_cppext does not import/run the C++ extension

### ISSUE THREAD

**Issue by encukou** [python/cpython#94751]
The `test_cppext` test case only checks that the C++ extension builds. It does not import and run test cases, so it can't catch runtime issues like #94731.

**Comment by vstinner** [comment]
> The test_cppext test case only checks that the C++ extension builds

Right, for now it's its only purpose as written at top of Lib/test/_testcppext.cpp:

```
// gh-91321: Very basic C++ test extension to check that the Python C API is
// compatible with C++ and does not emit C++ compiler warnings.
```

And Lib/test/test_cppext.py:

```
# gh-91321: Build a basic C++ test extension to check that the Python C API is
# compatible with C++ and does not emit C++ compiler warnings.
```

The reason is that I was too lazy to finish the work to actually import the built extension. I got enough C++ issues to handle. By the way, it seems like most of the work to fix C++ warnings has been reverted by commit 6cbb57f62d345d7a5d6aeb1b3b5d37a845344d5e.

I also got busy with replacing distutils with setuptools (using venv).
